File: mobile/flutter/lib_flutter_base/android/src/main/java/com/codesdancing/flutter/STFlutterBusHandler.kt
package com.codesdancing.flutter

import android.app.Application
import android.content.Context
import com.smart.library.STInitializer
import com.smart.library.util.bus.STBusManager

class STFlutterBusHandler : STBusManager.IBusHandler {

    override fun onInitOnce(application: Application?, callback: ((success: Boolean) -> Unit)?) {
        application ?: return
        STFlutterInitializer.initial(application, STInitializer.debug())
        callback?.invoke(true)
    }

    override fun onUpgradeOnce(application: Application?) {
    }

    /**
     * smart://template/flutter?page=demo&params=jsonString
     */
    override fun onCall(context: Context?, busFunctionName: String, vararg params: Any) {
        when (busFunctionName) {
            "flutter/open" -> {
                if (context != null) {
                    val schemaUrl:String = (params.getOrNull(0) as? String) ?: "smart://template/flutter?page=flutter_settings&params="
                    STFlutterUtils.openFlutterPageBySchema(schemaUrl)
                }
            }
        }
    }

    override fun onAsyncCall(callback: ((key: Any?, value: Any?) -> Unit)?, context: Context?, busFunctionName: String, vararg params: Any) {

    }
}

File: mobile/flutter/lib_flutter_base/ios/Classes/FlutterBase/STFlutterBusHandler.m
//
//  STFlutterBusHandler.m
//  lib_flutter_base
//
//  Created by krmao on 2021/3/9.
//

#import "STFlutterBusHandler.h"
#import "STFlutterUtils.h"
#import <LibIosBase/STSystemUtil.h>
#import <STFlutterInitializer.h>
#import <LibIosBase/STInitializer.h>

@interface STFlutterBusHandler()
@end

@implementation STFlutterBusHandler

- (id)initWithHost:(NSString *)host{
    if (self = [super initWithHost:host]) {
        NSLog(@"STFlutterBusHandler initWithHost host=%@", host);
        [[STFlutterInitializer sharedInstance] initial:[STInitializer sharedInstance].config.application debug:YES];
    }
    return self;
}

- (id)doDataJob:(NSString *)businessName params:(NSArray *)params{
    NSLog(@"STFlutterBusHandler doDataJob businessName=%@, params=%@", businessName, params);
    if ([businessName isEqual:@"flutter/open"] ) {
        NSString * schemaUrl = (NSString *)params.firstObject;
        if(schemaUrl == nil){
            schemaUrl= @"smart://template/flutter?page=flutter_settings&params=";
        }
        [STFlutterUtils openNewFlutterViewControllerBySchema:[STSystemUtil topViewController] schemaUrl:schemaUrl];
    }
    return NULL;
}

@end
